cedar-policy-validator/src/diagnostics/validation_errors.rs

```diff
@@ -448,6 +449,114 @@ impl Diagnostic for HierarchyNotRespected {
     }
 }
 
+#[derive(Debug, Clone, Hash, Eq, PartialEq, Error, Copy)]
+/// Represents how many entity dereferences can be applied to a node.
+/// `None` represents infinity.
+pub struct EntityDerefLevel {
+    /// `None` represents infinity.
+    pub level: Option<i64>,
+}
+
+impl Display for EntityDerefLevel {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        match self.level {
+            Some(l) => write!(f, "{l}"),
+            None => write!(f, "INF"),
```

Comment:
Is this a public display impl, if so we probably want a nicer display than `INF` (just `infinity`? `infinite`?)